prost-build/src/extern_paths.rs

```diff
@@ -27,27 +23,46 @@ pub struct ExternPaths {
 }
 
 impl ExternPaths {
-    pub fn new(paths: &[(String, String)], prost_types: bool)-> Result<ExternPaths, String> {
+    pub fn new(paths: &[(String, String)], prost_types: bool) -> Result<ExternPaths, String> {
         let mut extern_paths = ExternPaths {
             extern_paths: HashMap::new(),
         };
 
         for (proto_path, rust_path) in paths {
-            extern_paths.insert(proto_path.clone(), rust_path.clone())?;
+            if rust_path.starts_with("::") && cfg!(feature = "build-2018") {
+                extern_paths.insert(proto_path.clone(), format!("crate{}", rust_path))?;
```

Comment:
My gut is that converting `::` to `crate` is a bit too magical.  I think it should be the case that callers of `Config::extern_path` (in the respective build.rs) should know what the edition is, and should be able to switch between `::` and `crate::` appropriately.  WDYT?  It would definitely be worth documenting edition-2018 behavior in the method doc.